urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/main/Benchmark.java

```diff
@@ -0,0 +1,789 @@
+                    entry.getId(),
+                    entry.getMinDuration(),
+                    entry.getMaxDuration(),
+                    entry.getAllocation().toString(),
+                    entry.getDomain() == null ? "NA" : entry.getDomain().getBaseIntervals().toString()
+                    );
+            
+        }
+    }
+    
+    private static SolverState benchSolve(ProblemDefinition problem) {
+       
+        
+        // solve
+        Solver solver = new Solver();
+        solver.timeoutLimit = SOLVER_TIMEOUT;
+        solver.properties = solverProperties;
+        
+        SolverThread run = new SolverThread(solver, problem);
+        //run.solver.pause();
+        run.start();
+//        if (System.console() != null) {
+//            run.solver.pause();
+//        while(run.isAlive()) {
+//            String cmd = System.console().readLine();
+//            System.out.println(cmd);
+//            
+//            switch(cmd) {
+//                case "p":
+//                    run.solver.pause();
+//                    break;
+//                case "s":
+//                    System.out.println("Step");
+//                    run.solver.step();
+//                    break;
+//                case "r":
+//                    System.out.println("Printing state");
+//                    run.solver.printState();
+//                    break;
+//                case "q":
+//                    run.solver.timeoutLimit = 0;
+//                    run.solver.step();
+//                    break;
+//            }
+//        }
+//        }
+        try {
+            run.join();
+        } catch(Exception e) {
+            
+        }
+        
+        
+        return run.state;
+    }
+
+    
+    private static void printBestSolution(SolverState state) {
+        SolverSolution best = state.getBestSolution();
+        
+        // print solution
+        for (Map.Entry<Occurrence, OccurrenceAllocation> entry : best.getSchedule().getOccurrencesAllocations()) {
+            entry.getKey().setAllocation(entry.getValue());
+            System.out.printf("%s\ta:%s\tc:%s:%s:%s\n", 
+                    entry.getKey().getId(), 
+                    entry.getValue().toString(), //allocation
+                    entry.getKey().getPreferrenceCost(),
+                    entry.getKey().getDurationCost(),
+                    entry.getKey().getPreferredStartCost()
+                    );
+            
+        }
+        System.out.printf("Conflicting: %s, Cost: %s, Found in: %s", 
+                best.constraintsCost(), 
+                best.preferenceCost(), 
+                state.getLastBestIteration());
+        
+    }
+
+    private static void testProblem(ProblemDefinition p) {
+        long startTime = System.currentTimeMillis();
+        SolverState state = benchSolve(p);
+        System.out.println(String.format("END in %s, iterations %s", System.currentTimeMillis() - startTime, state.getItearation()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getItearation&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 499. Consider formatting this data with &#39;%d&#39; instead.